experimental/devices/mfrc522/mfrc522.go

```diff
@@ -0,0 +1,757 @@
+package mfrc522
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/spi"
+	"periph.io/x/periph/experimental/devices/mfrc522/commands"
+)
+
+// BlockAccess defines the block access bits.
+type BlockAccess byte
+
+// SectorTrailerAccess defines the sector trailing block access bits.
+type SectorTrailerAccess byte
+
+// Access bits.
+const (
+	AnyKeyRWID    BlockAccess = iota
+	RAB_WN_IN_DN              = 0x02 // Read (A|B), Write (None), Increment (None), Decrement(None)
+	RAB_WB_IN_DN              = 0x04
+	RAB_WB_IB_DAB             = 0x06
+	RAB_WN_IN_DAB             = 0x01
+	RB_WB_IN_DN               = 0x03
+	RB_WN_IN_DN               = 0x05
+	RN_WN_IN_DN               = 0x07
+
+	KeyA_RN_WA_BITS_RA_WN_KeyB_RA_WA        SectorTrailerAccess = iota
+	KeyA_RN_WN_BITS_RA_WN_KeyB_RA_WN                            = 0x02
+	KeyA_RN_WB_BITS_RAB_WN_KeyB_RN_WB                           = 0x04
+	KeyA_RN_WN_BITS_RAB_WN_KeyB_RN_WN                           = 0x06
+	KeyA_RN_WA_BITS_RA_WA_KeyB_RA_WA                            = 0x01
+	KeyA_RN_WB_BITS_RAB_WB_KeyB_RN_WB                           = 0x03
+	KeyA_RN_WN_BITS_RAB_WB_KeyB_RN_WN                           = 0x05
+	KeyA_RN_WN_BITS_RAB_WN_KeyB_RN_WN_EXTRA                     = 0x07
+)
+
+// AuthStatus indicates the authentication response, could be one of AuthOk, AuthReadFailure or AuthFailure
+type AuthStatus byte
+
+// Card authentication status enum.
+const (
+	AuthOk AuthStatus = iota
+	AuthReadFailure
+	AuthFailure
+)
+
+// BlocksAccess defines the access structure for first 3 blocks of the sector and the access bits for the
+// sector trail.
+type BlocksAccess struct {
+	B0, B1, B2 BlockAccess
+	B3         SectorTrailerAccess
+}
+
+// DefaultKey provides the default bytes for card authentication for method B.
+var DefaultKey = []byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
+
+// Dev is an handle to an MFRC522 RFID reader.
+type Dev struct {
+	resetPin         gpio.PinOut
+	irqPin           gpio.PinIn
+	operationTimeout time.Duration
+	spiDev           spi.Conn
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+//
+//							MFRC522 SPI Dev public API
+//
+///////////////////////////////////////////////////////////////////////////////////////////
+
+// NewSPI creates and initializes the RFID card reader attached to SPI.
+//
+// 	spiPort - the SPI device to use.
+// 	resetPin - reset GPIO pin.
+// 	irqPin - irq GPIO pin.
+func NewSPI(spiPort spi.Port, resetPin gpio.PinOut, irqPin gpio.PinIn) (*Dev, error) {
+
+	if resetPin == nil {
+		return nil, wrapf("reset pin is not set")
+	}
+	if irqPin == nil {
+		return nil, wrapf("IRQ pin is not set")
+	}
+
+	spiDev, err := spiPort.Connect(10000000, spi.Mode0, 8)
+	if err != nil {
+		return nil, err
+	}
+
+	dev := &Dev{
+		spiDev:           spiDev,
+		operationTimeout: 30 * time.Second,
+	}
+
+	if err := resetPin.Out(gpio.High); err != nil {
```

Comment:
Move these before constructing Dev, so that you can use:

```
dev := &Dev{
  spiDev:           spiDev,
  resetPin: resetPin,
  irqPin: irqPin,
  operationTimeout: 30 * time.Second,
}
```